### PR TITLE
fix(forms): Make some fixes to typed forms API and docs.

### PR DESCRIPTION
**Pull request by dylhunn**
* `FormRecord` JSDoc should now appear on a.io
* The `{@see foo#bar}` syntax previously did not work, and has been replace with backticks
* Value and RawValue have been removed from the public API (rolling back #45978); this needs more discussion (see #fw-forms)

### CODE REVIEW

File: packages/forms/src/model/form_group.ts [hunk 1]
```diff
@@ -629,49 +625,53 @@ export class FormRecord<TControl extends AbstractControl<Value<TControl>, RawVal
...
-   * {@see FormGroup#removeControl}
+   * See `FormGroup#removeControl`
    */
   removeControl(name: string, options?: {emitEvent?: boolean}): void;
 
   /**
    * Replace an existing control.
    *
-   * {@see FormGroup#setControl}
+   * See `FormGroup#setControl`
```
Comment (AndrewKushnir) [inline packages/forms/src/model/form_group.ts:657]: nit: I'd propose updating this line (and similar ones in this PR) a bit:
```suggestion
   * See `FormGroup#setControl` for additional information.
```

**Review by AndrewKushnir — COMMENTED** [review]
@dylhunn thanks for the updates 👍 I've left one minor comment.

I also think that we should probably have 2 separate commits in this PR: one is a revert of the previous commit that made the `Value` and `RawValue` public and the second one is the docs update (and make it `docs(forms): ...`, so it's not included in the changeog). That should help the tooling to ensure that the initial "feat(forms)" commit is not present in a changelog (or at least that'd be visible to a caretaker would would assemble the changelog for v14.0.0 release).

**Review by AndrewKushnir — APPROVED** [review]
Thanks for addressing the feedback 👍 

Could you please also change the `fix(forms):` -> `docs(forms):` in the commit message?

**Comment by dylhunn** [comment]
Done, thanks!

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by alxhub — APPROVED** [review]
Reviewed-for: public-api

**Comment by dylhunn** [comment]
merge-assistance: failures are unrelated

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 038ba8adeae833135c9ba068cae3aa3859016272.

**Pull request closed without merging**